plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeTransformTelemetryService.kt

```diff
@@ -0,0 +1,23 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codemodernizer.model
+
+import com.intellij.openapi.components.Service
+
+@Service(Service.Level.APP)
+public final class CodeTransformTelemetryService {
```

Comment:
@rli Should I drop the the instance and getInstance and just have my metadata a member of my telemetry class?

It looks like the service [docs](https://plugins.jetbrains.com/docs/intellij/plugin-services.html#t2t7gz_184) are implying that it will find the service level singelton if existed when calling my class like
```
service<CodeTransformTelemetryService>()
```

<img width="903" alt="Screenshot 2024-06-11 at 4 05 26 PM" src="https://github.com/aws/aws-toolkit-jetbrains/assets/38665452/9f595d26-b29b-421a-a90c-495f63c28059">

Reply:
`instance` for sure
`getInstance` is just a convenience to save needing to import two things every time. you are the only consumer of the service so it can go either way